### PR TITLE
🔨 REFACTOR (props): move prop declaration into view component

### LINKED ISSUE #29: Inconsistent placements of props data structures
## Description

In the project, both `props` and `model` data structures are separated. However, the declaration for `props` do not follow a consistent location. 

In `Project/`, `ProjectProps`'s declaration can be found in an isolated file (`ProjectProps.ts`) in the `models/` directory. This however is not the case for `Project/Rows/` as their `props` are declared within its respective view component. E.g. `ProjectRowImageProps` is declared in `ProjectRowImage.tsx`.

## Issue

This does not cause any problems when building, but as the project grows it will get confusing where things should be. 

## Next

I would like to get a sense from the team for which structure they think is better.

### PR DESCRIPTION
**Pull request by lim-james**
## Description

As suggested, `props` will be more tightly coupled with their respective components. Hence, `props` that were declared in a separate file, are now shifted into their view's file.

Resolves #29

### CODE REVIEW

**Comment by lim-james** [comment]
@sophiewongsp I can't request changes as I was the one who created the PR, so I will be typing the revisions here

**Comment by lim-james** [comment]
@sophiewongsp 

## Issue

`SectionProps` and `RoleProps` are still separately declared. 

## Description

I understand that you tried to move them in  [147559d](https://github.com/niji-co/Fusion-Web/pull/36/commits/147559dda5a24dab3049993119d54261be5e41c6) and [d74fb4d](https://github.com/niji-co/Fusion-Web/pull/36/commits/d74fb4dd6bf3c8c58f9390eefe5db1b29294a7d0), but encountered the following build errors. 

```
~/Fusion-Web/src/pages/Exhibitions/Role/index.ts
...
  2:38  error  Missing file extension for "./models/RoleProps"        import/extensions
  2:38  error  Unable to resolve path to module './models/RoleProps'  import/no-unresolved

~/Fusion-Web/src/pages/Exhibitions/Section/index.ts
...
  2:26  error  Missing file extension for "./models/SectionProps"        import/extensions
  2:26  error  Unable to resolve path to module './models/SectionProps'  import/no-unresolved
```

These are the same errors, and for the case of the `Section/index.ts` this was the line that caused it.
```ts
// Section/index.ts
import SectionProps from "./models/SectionProps";
```
You had deleted the file and was still trying to import from it.

## Plausible solution

A straightforward, _but inadvisable,_ fix would have been this.
1. Export props declaration from the component script
```ts
// Section.tsx
interface SectionProps {
  sectionName: string;
  exhibitor: string;
}

// ... implementation of view component

export default Section;    // Existing export of view component
export { SectionProps };   // New - Export your props together too
```
2. Import it in `index`, replacing the broken line
```ts
// Section/index.ts
import Section, { SectionProps } from "./Section";

// Removing this line
// import SectionProps from "./models/SectionProps";

export default Section;
export { SectionProps };
```

> Even though this solution works, I would not recommend it. The purpose of this PR is to hide away the props, which is the opposite of what these changes would be doing.

## Props vs Model - Solution

A `model` contains purely data, either retrieved from the server or entered by the user. However, when creating a view we sometimes use [HTML attributes](https://reactjs.org/docs/dom-elements.html#all-supported-html-attributes) - `className`, `style`, etc., all of which should not be a part of the `model`. This is the purpose of `props` - it represents the information to be passed in when creating a view, so this includes both your [HTML attributes](https://reactjs.org/docs/dom-elements.html#all-supported-html-attributes) and your `model`. 

### Example 
```ts
// UserModel.ts
// A pure model, only containing the needed data.
interface UserModel {
  username: string;
  password: string;
}

export default UserModel;
```
```ts
// User.tsx - user view component
import { HTMLAttributes } from "react";

// Props with both model's data and HTML attributes
type UserProps = UserModel & HTMLAttributes<HTMLElement>; 
// or 
interface UserProps extends UserModel, HTMLAttributes<HTMLElement> {} 
// The latter is inadvisible, empty interfaces should be replaced with type.
// This style should only be used if the props has their own attributes.

// ... view declaration
// no exports for props
```

In this use case, the `model`'s declaration can be exposed while the declaration of `props` is hidden away.

**Comment by lim-james** [comment]
## Issue 

`ExhbitionProps` is a `model`, `props` are to extend `HTMLAttributes`. (Refer to above comment to understand the difference between `props` vs `model`.

File: src/AppRoutes.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const AppRoutes = (): ReactElement => (
       <Route path="/forgot" component={ForgotPassword} />
       <Route path="/profile" component={Profile} />
 
-      <Route exact path="/exhibitions" component={Exhibitions} />
+      <Route exact path="/exhibitions" component={Exhibition} />
```
Comment (lim-james) [inline src/AppRoutes.tsx:26]:
```ts
path="/exhibitions"
```
Typo, should be `exhibition`

**Review by sophiewongsp — APPROVED** [review]
Everything completed :)

**Pull request merged**